### ISSUE TITLE
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd0 in position 20: ordinal not in range(128)

### ISSUE THREAD

**Issue by mgedmin** [jazzband/pip-tools#224]
My requirements.in looks like this:

```
pip-tools
blogofile
ghp-import2
watchdog
argcomplete
linkchecker
```

I'm running `pip-compile` on Python 2.7.8, and I get this:

```
Traceback (most recent call last):
  File "bin/pip-compile", line 11, in <module>
    sys.exit(cli())
  File "/home/mg/src/ivija-blogofile/.venv/local/lib/python2.7/site-packages/click/core.py", line 700, in __call__
    return self.main(*args, **kwargs)
  File "/home/mg/src/ivija-blogofile/.venv/local/lib/python2.7/site-packages/click/core.py", line 680, in main
    rv = self.invoke(ctx)
  File "/home/mg/src/ivija-blogofile/.venv/local/lib/python2.7/site-packages/click/core.py", line 873, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/mg/src/ivija-blogofile/.venv/local/lib/python2.7/site-packages/click/core.py", line 508, in invoke
    return callback(*args, **kwargs)
  File "/home/mg/src/ivija-blogofile/.venv/local/lib/python2.7/site-packages/piptools/scripts/compile.py", line 111, in cli
    results = resolver.resolve()
  File "/home/mg/src/ivija-blogofile/.venv/local/lib/python2.7/site-packages/piptools/resolver.py", line 78, in resolve
    has_changed, best_matches = self._resolve_one_round()
  File "/home/mg/src/ivija-blogofile/.venv/local/lib/python2.7/site-packages/piptools/resolver.py", line 162, in _resolve_one_round
    for best_match in best_matches
  File "/home/mg/src/ivija-blogofile/.venv/local/lib/python2.7/site-packages/piptools/resolver.py", line 163, in <genexpr>
    for dep in self._iter_dependencies(best_match))
  File "/home/mg/src/ivija-blogofile/.venv/local/lib/python2.7/site-packages/piptools/resolver.py", line 234, in _iter_dependencies
    dependencies = self.repository.get_dependencies(ireq)
  File "/home/mg/src/ivija-blogofile/.venv/local/lib/python2.7/site-packages/piptools/repositories/pypi.py", line 128, in get_dependencies
    dependencies = reqset._prepare_file(self.finder, ireq)
  File "/home/mg/src/ivija-blogofile/.venv/local/lib/python2.7/site-packages/pip/req/req_set.py", line 491, in _prepare_file
    session=self.session)
  File "/home/mg/src/ivija-blogofile/.venv/local/lib/python2.7/site-packages/pip/download.py", line 825, in unpack_url
    session,
  File "/home/mg/src/ivija-blogofile/.venv/local/lib/python2.7/site-packages/pip/download.py", line 677, in unpack_http_url
    unpack_file(from_path, location, content_type, link)
  File "/home/mg/src/ivija-blogofile/.venv/local/lib/python2.7/site-packages/pip/utils/__init__.py", line 650, in unpack_file
    untar_file(filename, location)
  File "/home/mg/src/ivija-blogofile/.venv/local/lib/python2.7/site-packages/pip/utils/__init__.py", line 596, in untar_file
    path = os.path.join(location, fn)
  File "/home/mg/src/ivija-blogofile/.venv/lib/python2.7/posixpath.py", line 73, in join
    path += '/' + b
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd0 in position 20: ordinal not in range(128)
```

**Comment by mgedmin** [comment]
If I comment out `linkchecker`, the error goes away.

**Comment by kylios** [comment]
Any progress on this issue? I'm seeing the same error caused by the package `scandir == 1.4`

**Comment by last-partizan** [comment]
https://github.com/benhoyt/scandir/issues/72

here is related issue for scandir, probably this will be fixed with next scandir update

**Comment by mlazowik** [comment]
@kylios #433

**Comment by vphilippon** [comment]
Should be fixed by #568, closing.
